examples/platform/silabs/efr32/rs911x/hal/sl_si91x_ncp_utility.c

```diff
@@ -80,21 +82,14 @@ SemaphoreHandle_t spi_sem_sync_hdl;
 sl_status_t sl_wfx_host_pre_lcd_spi_transfer(void)
 {
 #if SL_SPICTRL_MUX
-    if (sl_wfx_host_spi_cs_deassert() != SL_STATUS_OK)
-    {
-        return SL_STATUS_FAIL;
-    }
     xSemaphoreTake(spi_sem_sync_hdl, portMAX_DELAY);
 #endif // SL_SPICTRL_MUX
-    // sl_memlcd_refresh takes care of SPIDRV_Init()
-    if (SL_STATUS_OK != sl_memlcd_refresh(sl_memlcd_get()))
+    if (SL_STATUS_OK != sl_board_enable_display())

```

Comment:
Release semaphore before return for failure case.
Check other places also.

Reply:
done @senthilku